sdcm/utils/common.py

```diff
@@ -1513,19 +1513,21 @@ def get_branched_ami(scylla_version: str, region_name: str, arch: AwsArchType =
         {"Name": "tag:branch", "Values": [branch, ], },
         {"Name": "architecture", "Values": [arch, ], },
     ]
-    if build_id not in ("latest", "all",):
-        filters.append({'Name': 'tag:build-id', 'Values': [build_id, ], })
 
     LOGGER.info("Looking for AMIs match [%s]", scylla_version)
     ec2_resource: EC2ServiceResource = boto3.resource("ec2", region_name=region_name)
-    images = sorted(
-        ec2_resource.images.filter(Filters=filters),
-        key=lambda x: x.creation_date,
-        reverse=True,
-    )
+    if build_id not in ("latest", "all",):
+        images = [
+            ec2_resource.images.filter(Filters=filters + [{'Name': 'tag:build-id', 'Values': [build_id, ], }]),
+            ec2_resource.images.filter(Filters=filters + [{'Name': 'tag:build_id', 'Values': [build_id, ], }]),
```

Comment:
why are you doing twice same request?

Reply:
Not the same, unfortunately: build-id and build_id. Hello, releng, we love you.

Reply:
ah, missed that detail (and i saw it in hydra calls few times already)